CRM/Member/BAO/Membership.php

```diff
@@ -139,38 +139,65 @@ public static function add(&$params, $ids = array()) {
     // reset the group contact cache since smart groups might be affected due to this
     CRM_Contact_BAO_GroupContactCache::opportunisticCacheFlush();
 
+    $activityParams = array();
+    $allStatus = CRM_Member_BAO_Membership::buildOptions('status_id', 'get');
+    if (!empty($params['is_pay_later']) ||
```

Comment:
I guess one thing we should note is that in 4.7 we made it so that the forms should always create Pending contributions & then use the completetransaction api to complete them when the payment is complete (by whatever method). The activity creation code was never updated to reflect that, which I think it making it confusing to think through.

I think we can probably create scheduled activities from that form on the basis that the form should follow the 4.7 flow?

So, what about the membership BAO accepts a parameter 'membership_activity_status_id' => 'Scheduled' & we set it to that when coming in off that form flow & it can default to Completed in other scenarios (at least for now) ?